modules/control/submodules/mpc_controller_submodule.cc

```diff
@@ -69,10 +70,13 @@ bool MPCControllerSubmodule::Proc(
   ADEBUG << "MPC controller submodule started ....";
 
   // skip produce control command when estop for MPC controller
-  if (preprocessor_status->estop()) {
-    // recording estop reason
+  StatusPb pre_status = preprocessor_status->header().status();
+  if (pre_status.error_code() != ErrorCode::OK) {
+    control_core_command.mutable_header()->mutable_status()->set_error_code(
+        pre_status.error_code());
     control_core_command.mutable_header()->mutable_status()->set_msg(
-        preprocessor_status->estop_reason());
+        pre_status.msg());
```

Comment:
So `control_core_command.mutable_header()->mutable_status()->CopyFrom(pre_status)`?

Reply:
Updated. Thank you.